### PR TITLE
Support incremental protobuf compilation and remove recursive proto f…

### PR DESCRIPTION
**Pull request by pvallem**
…iles compilation

Summary: protoc maven jar, is by defualt compiling all proto files recursively. Made sure it doesn't compile recursively.
This PR is exactly same as the #2, just on top tagv3.1.0.2 This version they fixed few issues.

Test Plan: NA

Reviewers: @vjeyakum @andy256 @gshashitetration @sunigup2

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <groupId>com.github.os72</groupId>
   <artifactId>protoc-jar-maven-plugin</artifactId>
   <packaging>maven-plugin</packaging>
-  <version>3.1.0.2</version>
+  <version>3.1.0.2-tet-SNAPSHOT</version>
```
Comment (ghost) [inline pom.xml]: Could you not make it a -SNAPSHOT version, so we don't have to keep downloading it every time we build?

**Review by ghost — CHANGES_REQUESTED** [review]
Ugh, also their original code uses tabs and this patch has spaces.

Other than that, I assume this is pretty much the same as what we've been using, and that seems to work pretty well.

File: pom.xml (hunk 1 above)
Comment (pvallem) [reply pom.xml]: sure. Will remove -SNAPSHOT

Comment (pvallem) [reply pom.xml]: Done.

**Comment by pvallem** [comment]
MyBad, i configured my vim to use spaces... :) Is that ok or shall i change it to tab and resubmit?

**Review by ghost — APPROVED** [review]

**Pull request merged**